Unexpected error in module orderfind.py

Hi, I was running a dataset that from [SRR15275213](https://www.ncbi.nlm.nih.gov/sra/?term=SRR15275213), but got an assert error in runtime. So i modified source code after assert to output values.

**dataset preprocess**
1. assembly by using hifiasm-meta with default parameters
  - `hifiasm_meta -t 16 -o asm SRR15275213.fastq.gz`
  - `gfatools gfa2fa asm.p_ctg.gfa > human-p-ctg.fa` 

3. run antismash locally with `antismash --genefinding-tool prodigal-m --output-dir human-hifiasm-meta --allow-long-headers -c 16 human-p-ctg.fa --logfile 1.log 2>>1.log`
log file as following

**Error exception**
```bash
Traceback (most recent call last):
  File "/home/dwpeng/github-source/antismash/.venv/bin/antismash", line 33, in <module>
    sys.exit(load_entry_point('antismash==8.dev0', 'console_scripts', 'antismash')())
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/home/dwpeng/github-source/antismash/.venv/lib/python3.13/site-packages/antismash/__main__.py", line 127, in entrypoint
    sys.exit(main(sys.argv[1:]))
             ~~~~^^^^^^^^^^^^^^
  File "/home/dwpeng/github-source/antismash/.venv/lib/python3.13/site-packages/antismash/__main__.py", line 116, in main
    return antismash.run_antismash(sequence, options)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
  File "/home/dwpeng/github-source/antismash/.venv/lib/python3.13/site-packages/antismash/main.py", line 683, in run_antismash
    result = _run_antismash(sequence_file, options)
  File "/home/dwpeng/github-source/antismash/.venv/lib/python3.13/site-packages/antismash/main.py", line 750, in _run_antismash
    analysis_timings = analyse_record(record, options, get_analysis_modules(), module_results)
  File "/home/dwpeng/github-source/antismash/.venv/lib/python3.13/site-packages/antismash/main.py", line 303, in analyse_record
    run_module(record, module, options, previous_result, timings)
    ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dwpeng/github-source/antismash/.venv/lib/python3.13/site-packages/antismash/main.py", line 275, in run_module
    results = module.run_on_record(record, results, options)
  File "/home/dwpeng/github-source/antismash/.venv/lib/python3.13/site-packages/antismash/modules/nrps_pks/__init__.py", line 97, in run_on_record
    specific_analysis(record, results, options)
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/dwpeng/github-source/antismash/.venv/lib/python3.13/site-packages/antismash/modules/nrps_pks/specific_analysis.py", line 68, in specific_analysis
    candidate_cluster_predictions = analyse_biosynthetic_order(nrps_pks_genes, results.consensus, record)
  File "/home/dwpeng/github-source/antismash/.venv/lib/python3.13/site-packages/antismash/modules/nrps_pks/orderfinder.py", line 116, in analyse_biosynthetic_order
    geneorder = find_colinear_order(list(with_complete))
  File "/home/dwpeng/github-source/antismash/.venv/lib/python3.13/site-packages/antismash/modules/nrps_pks/orderfinder.py", line 493, in find_colinear_order
    assert module is second.modules[0], str(first) + " " + str(second)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: CDS(ctg201_293, [311512:318007](-)) CDS(ctg201_296, [320172:322908](-))
```

--- a/antismash/modules/nrps_pks/orderfinder.py
+++ b/antismash/modules/nrps_pks/orderfinder.py
@@ -472,12 +472,13 @@ def find_colinear_order(cds_features: List[CDSFeature]) -> List[CDSFeature]:
     subgroups = [[cds_features[0]]]
     current_subgroup = subgroups[-1]
 
-    direction = 0
+    direction = cds_features[0].strand
     for gene in cds_features[1:]:
         direction += gene.strand
         # if the first module of a cds has a cross-cds module, then
         # the previous CDS' last module is also cross-cds module, regardless of direction
-        if gene.modules and len(gene.modules[0].parent_cds_names) > 1:
+        cross_cds = gene.modules and gene.modules[0].is_multigene_module()
+        if cross_cds and gene.modules[0] is current_subgroup[-1].modules[-1]:
             current_subgroup.append(gene)
         else:  # no need to check the end, since the next CDS will be found at the start
             subgroups.append([gene])

--- a/antismash/modules/nrps_pks/test/test_orderfinder.py
+++ b/antismash/modules/nrps_pks/test/test_orderfinder.py
@@ -400,6 +400,51 @@ class TestOrdering(unittest.TestCase):
         expected = [cds.get_name() for cds in [chain_c, chain_b, chain_a, other, terminal]]
         assert [cds.get_name() for cds in orderfinder.find_colinear_order(cdses)] == expected
 
+    def test_colinear_chain_single_multigene(self):
+        # if a gene has a single module which is multi-gene, it must connect to the correct subgroup
+
+        def set_qualifier(cds, names):
+            cds.nrps_pks = DummyNRPSQualfier()
+            cds.nrps_pks.domain_names = names
+
+        pks = ["PKS_KS", "PKS_AT", "KR", "ACP"]
+        nrps = ["Condensation", "AMP-binding", "PP-binding"]
+
+        other_names = [f"simple_{i}" for i in range(3)]
+        others = []
+        for i, name in enumerate(other_names):
+            simple = DummyCDS(30 * i, 30 * (i  + 1), strand=-1, locus_tag=name)
+            for _ in range(2):
+                module = DummyModule(domains=nrps, complete=True,
+                                     location=simple.location.clone_with_offset(5))
+                simple.add_module(module)
+                simple.modules[-1]._parent_cds_names = [name]
+            set_qualifier(simple, nrps * 2)
+            others.append(simple)
+
+        first_half = DummyCDS(201, 400, locus_tag="A")
+        module = DummyModule(location=FeatureLocation(300, 500, 1),
+                             domains=pks, complete=True, iterative=True)
+        module._parent_cds_names = ["A", "B"]
+        first_half.add_module(module)
+        set_qualifier(first_half, pks[:2])
+
+        second_half = DummyCDS(401, 600, locus_tag="B")
+        second_half.add_module(module)
+        set_qualifier(second_half, pks[2:])
+
+        cdses = others + [first_half, second_half]
+
+        expected = ["A", "B"] + list(reversed(other_names))
+        assert [cds.get_name() for cds in orderfinder.find_colinear_order(cdses)] == expected
+
+        # and the reverse strand for the multigene module
+        first_half.location = FeatureLocation(201, 400, -1)
+        second_half.location = FeatureLocation(401, 600, -1)
+        module._parent_cds_names.reverse()
+        expected = ["B", "A"] + list(reversed(other_names))
+        assert [cds.get_name() for cds in orderfinder.find_colinear_order(cdses)] == expected
+
 
 class TestEnzymeCounter(unittest.TestCase):
     def run_finder(self, names, modules_by_cds):

